templates/base/server/.rtag/logstore.ts.hbs

```diff
@@ -0,0 +1,40 @@
+import fs from "fs";
+import path from "path";
+
+export default class LogStore {
+  private storageDir: string;
+  private partitionFiles: Map<string, number>;
+
+  public constructor(storageDir: string) {
+    this.storageDir = storageDir;
+    this.partitionFiles = new Map();
+  }
+
+  public append(partitionId: string, time: number, record: Uint8Array) {
+    if (!this.partitionFiles.has(partitionId)) {
+      this.partitionFiles.set(partitionId, fs.openSync(path.join(this.storageDir, partitionId), "a"));
+    }
+    const fd = this.partitionFiles.get(partitionId)!;
+
+    const buf = Buffer.alloc(12);
+    buf.writeBigUInt64LE(BigInt(time));
+    buf.writeUInt16LE(record.byteLength, 8);
+    fs.appendFileSync(fd, buf);
```

Comment:
Why not append both `buf` and `record` together in one synchronous call? If the server is stopped after appending `buf` but not `record`, you kinda have an invalid/corrupt record.

Reply:
done